### PR TITLE
bpo-30512: add CAN Socket support for NetBSD

### PR DESCRIPTION
**Pull request by 0-wiz-0**
This also fixes a small unrelated bug in the configure script - test(1) only supports "=" as a comparison operator, "==" is a bash extension. If this should be a separate pull request, please let me know.

https://bugs.python.org/issue30512

### CODE REVIEW

**Comment by taleinat** [comment]
> This also fixes a small unrelated bug in the configure script - test(1) only supports "=" as a comparison operator, "==" is a bash extension. If this should be a separate pull request, please let me know.

Indeed, it would be better for that to be a separate PR. Thanks!

**Comment by taleinat** [comment]
@serhiy-storchaka, since according to the b.p.o. issue you've worked on this in the past, it would be good to get your review of this!

**Comment by 0-wiz-0** [comment]
@taleinat I've made a separate pull request for the configure script fix, but I don't know how to mark it as "skip news", please advise. Thanks!

**Comment by taleinat** [comment]
> @taleinat I've made a separate pull request for the configure script fix, but I don't know how to mark it as "skip news", please advise. Thanks!

Thanks for doing that! Don't worry about the skip-news labels; only triagers and core devs can add those, but a mention in a comment is great :+1:

File: Modules/socketmodule.c
Comment (serhiy-storchaka) [inline Modules/socketmodule.c]: Is it not defined on NetBSD? Then after merging with the HAVE_LINUX_CAN_RAW_H case you can add `#ifdef CAN_RAW_ERR_FILTER`.

File: Modules/socketmodule.c
Comment (serhiy-storchaka) [inline Modules/socketmodule.c]: Why not merge this with blocks for HAVE_LINUX_CAN_H and HAVE_LINUX_CAN_RAW_H?

File: Misc/NEWS.d/next/Core and Builtins/2021-12-12-00-49-19.bpo-30512.nU9E9V.rst
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Core and Builtins/2021-12-12-00-49-19.bpo-30512.nU9E9V.rst]:
```suggestion
Add CAN Socket support for NetBSD.
```

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Please update the documentation for the socket module. Update `availability` directives, add `versionchanged` directives.

Add also an entry in the What's New file.

**Comment by 0-wiz-0** [comment]
Thanks for the review, @serhiy-storchaka !
I've tried addressing all your comments. I'm not sure what you mean by
```
Add also an entry in the What's New file.
```
since I thought that's what the blurb is used for.

**Comment by 0-wiz-0** [comment]
I have made the requested changes; please review again

File: Modules/socketmodule.c
Comment (serhiy-storchaka) [inline Modules/socketmodule.c]:
```suggestion
#ifdef CAN_RAW_ERR_FILTER
```
For consistency.

File: Doc/library/socket.rst
Comment (serhiy-storchaka) [inline Doc/library/socket.rst:399]: Are any of CAN_RAW_FD_FRAMES, CAN_RAW_JOIN_FILTERS, CAN_ISOTP, CAN_J1939 defined on NetBSD?

**Comment by serhiy-storchaka** [comment]
Blurb is used for creating full Changelog which includes all user-visible changed as they are made, including bugfixes.

There is a separate file `Doc/whatsnew/3.11.rst`. It is structured and only includes information about new features, deprecations and removals. In section "Improved Modules" add a subsection for "socket" and a description of the feature. E.g.:
```
socket
------
* Add CAN Socket support for NetBSD.
  (Contributed by Thomas Klausner in :issue:`30512`.)

```

File: Modules/socketmodule.c
Comment (0-wiz-0) [reply Modules/socketmodule.c]: Changed as suggested. (A couple lines above that, it's `#if defined` - the file is not very consistent :) ).

File: Doc/library/socket.rst
Comment (0-wiz-0) [reply Doc/library/socket.rst:399]: No -- I had checked that, these symbols do not exist on NetBSD.

**Comment by 0-wiz-0** [comment]
I've added a Whatsnew entry as suggested - thanks for the explanation & review!

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**
